### PR TITLE
Add new type of dashboard initial state for the built in and settings based dashboards

### PR DESCRIPTION
**Pull request by vovakulikov**
### Context

This PR improves our initial state of empty insights dashboard. Currently, on main we have the following empty dashboard state for all types of dashboards (virtual, built-in like personal and org dashboards, custom type of dashboards that were created by users) 

image

This button `create new insight` actually makes sense for built-in dashboards (the only one way to add insight to a built-in dashboard is to create a dashboard with some specific to this dashboard visibility settings)

But this `create new insight` button may confuse users on custom dashboards, cause just to create a code insight isn't enough to see this insight on some custom dashboards. In fact, you have to go to the dashboard page pick the custom dashboard and by dashboard context menu action `Add insights` add some insight to this custom dashboard.

So I assume it would make sense to have a different initial state for the custom dashboards. This PR add this specific initial state. 

image

By click on add insight, you will see the Add insight to dashboard modal UI 

Screenshot 2021-07-16 at 12 21 23

And in this UI you can find and add some insights to the active dashboard.

### CODE REVIEW

File: client/web/src/insights/pages/dashboards/dashboard-page/hooks/use-dashboard-permissions.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    switch (permissions.reason) {
+        case DashboardReasonDenied.UnknownDashboard:
+            return "We didn't find a dashboard. You can't edit unknown dashboard"
+        case DashboardReasonDenied.PermissionDenied:
+            return "You don't a permission to edit this dashboard"
+        case DashboardReasonDenied.BuiltInCantBeEdited:
+            return "This dashboard is built-in. You can't edit built-in dashboards"
+    }
```
Comment (eseliger) [inline client/web/src/insights/pages/dashboards/dashboard-page/hooks/use-dashboard-permissions.ts:91]: That's a pretty cool pattern, adding a reason to a simple bool value/

File: client/web/src/insights/pages/dashboards/dashboard-page/hooks/use-dashboard-permissions.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        return
+    }
+
+    switch (permissions.reason) {
+        case DashboardReasonDenied.UnknownDashboard:
+            return "We didn't find a dashboard. You can't edit unknown dashboard"
+        case DashboardReasonDenied.PermissionDenied:
+            return "You don't a permission to edit this dashboard"
+        case DashboardReasonDenied.BuiltInCantBeEdited:
+            return "This dashboard is built-in. You can't edit built-in dashboards"
```
Comment (eseliger) [inline client/web/src/insights/pages/dashboards/dashboard-page/hooks/use-dashboard-permissions.ts:90]: This sentence feels a bit redundant

```suggestion
            return "Built-in dashboards can't be edited"
```

File: client/web/src/insights/pages/dashboards/dashboard-page/hooks/use-dashboard-permissions.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+export function getTooltipMessage(permissions: DashboardPermissions): string | undefined {
+    if (permissions.isConfigurable) {
+        return
+    }
+
+    switch (permissions.reason) {
+        case DashboardReasonDenied.UnknownDashboard:
+            return "We didn't find a dashboard. You can't edit unknown dashboard"
```
Comment (eseliger) [inline client/web/src/insights/pages/dashboards/dashboard-page/hooks/use-dashboard-permissions.ts:86]: This sounds a bit strange to me, "not found, something that's not found is not editable". How about just something like

```suggestion
            return "Dashboard not found"
```

to me that implies that it also cannot be edited.

**Review by eseliger — APPROVED** [review]
I realize these error messages existed before, so feel free to ignore my comments :)

**Review by tjkandala — APPROVED** [review]

**Comment by AlicjaSuska** [comment]
Hi @vovakulikov Thank you for bringing this up. It's a good idea to change the flow. I'd like to change the texts slightly:
 'Assign insights' instead of 'Add insight'

**Pull request merged**